Firebase

CagriKIRT opened the pull request:
#What
Display "title" data from HackerNews realtime database to the project, issue #4 

#Why
To Learn how real time database works(issue #1 ), how to retrieve data, how to access database using firebase methods.

#How
First we implement database reference to a variable named "query" in "NewStoriesAdapter" class.
After that we request the title of the news by requesting data on the child. I attach "addListenerForSingleValueEvent" to the "query" for listening value changes and bind them to UI under "onBindViewHolder" function.

#Screenshot
![resim_2020-12-26_200200](https://user-images.githubusercontent.com/53116879/103155972-37750280-47b5-11eb-8a58-3026eab11c77.png)


#Anything Else?
I haven't delete the "HnData" data class. It will be useful in the future when we want to download other data
Thanks for the support 👍

CagriKIRT committed:
Add implementations for realtime database

safaorhan committed:
Get FirebaseRecyclerAdapter started

CagriKIRT committed:
Display news title from Hacker News firebase realtime database

safaorhan commented on app/src/main/java/school/cactus/hnmobile/NewStoriesAdapter.kt:
```diff
@@ -0,0 +1,52 @@
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val children = snapshot.getValue<String>()
+
+                storyItemHolder.textView.text = "News: "+children.toString()
```
We should not forget to format our code before commiting.

Actually there is a setting for this in Android Studio.

<img width="271" alt="Screen Shot 2020-12-27 at 16 43 44" src="https://user-images.githubusercontent.com/4990386/103172178-b07e6380-4862-11eb-9c85-10cf964c8f45.png">

CagriKIRT replied:
thanks, noted.

safaorhan commented on app/src/main/java/school/cactus/hnmobile/NewStoriesAdapter.kt:
```diff
@@ -0,0 +1,52 @@
+            }
+
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val children = snapshot.getValue<String>()
```
The name children is somehow misleading because it's plural but `snapshot.getValue<String>()` returns only a single String.

And also I am a bit confused now and didn't understand how you get the title only 🤔

CagriKIRT replied:
- I didn't notice. I will change the name, named "newstitle",

-when I request data from 

val queryStory = query.child("/v0/item/$model")
val newsTitle = snapshot!!.children

response is:

![screenshot](https://user-images.githubusercontent.com/53116879/103173299-1a027000-486b-11eb-8d63-38013e8f455c.png)

Because of that I enter child name "title" to get value from.

val queryStory = query.child("/v0/item/$model").child("title")
val newsTitle = snapshot.getValue<String>()

safaorhan commented on app/src/main/java/school/cactus/hnmobile/NewStoriesAdapter.kt:
```diff
@@ -0,0 +1,52 @@
+            override fun onDataChange(snapshot: DataSnapshot) {
+                val children = snapshot.getValue<String>()
+
+                storyItemHolder.textView.text = "News: "+children.toString()
```
if `children` is a String why use `children.toString()`?

safaorhan reviewed (changes requested):
Looking good but some clarification needed.

CagriKIRT committed:
Display news title from Hacker News firebase realtime database

safaorhan committed:
Refactor code for better naming.

safaorhan committed:
Add memory cache and disk cache

safaorhan commented:
You can merge after a final review.

CagriKIRT committed:
Display News from title

CagriKIRT commented:
## What?

Added bindTo function inside StoryItemHolder class
Added another activity and webview to display news

## Why?

With this user can enter any news. Its great exercise for learn how to  implement something to existing code

## How?

With bindTo function implemented inside StoryItemHolder, I can take "ur"l address in string format inside onBindViewHolder.
I use ıntent to send "url" address to NewsStoryActivity and displayed with webview

## Screenshot

![image](https://user-images.githubusercontent.com/53116879/104045434-9b54ed80-51ef-11eb-9cae-96a36a96c2f2.png)

## Anything Else?

I found the solution in another project on internet. If there is a better way to do it please let me know.
